src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs

```diff
@@ -161,12 +186,48 @@ private void AppendPath(ODataPath path)
 
                 case ODataPathKind.Operation:
                 case ODataPathKind.OperationImport:
+                    if (kind == ODataPathKind.Operation)
+                    {
+                        foreach (var kvp in _dollarCountPaths)
+                        {
+                            ODataPath dollarCountPath = kvp.Value.FirstOrDefault(p => DollarCountAndOperationPathsSimilar(p, path));
+                            if (dollarCountPath != null &&
+                                _allNavigationSourcePaths.TryGetValue(kvp.Key, out IList<ODataPath> dollarPathList))
+                            {
+                                dollarPathList.Remove(dollarCountPath);
+                                break;
+                            }
+                        }
+                    }
+
                     _allOperationPaths.Add(path);
                     break;
 
                 default:
                     return;
             }
+
+            bool DollarCountAndOperationPathsSimilar(ODataPath path1, ODataPath path2)
+            {
+                string countIdentifier = "count";
+
+                if ((path1.Kind == ODataPathKind.DollarCount && 
+                    path2.Kind == ODataPathKind.Operation && path2.LastSegment.Identifier.Equals(countIdentifier, StringComparison.OrdinalIgnoreCase)) ||
+                    (path2.Kind == ODataPathKind.DollarCount &&
+                    path1.Kind == ODataPathKind.Operation && path1.LastSegment.Identifier.Equals(countIdentifier, StringComparison.OrdinalIgnoreCase)))
```

Comment:
```suggestion
                    path1.Kind == ODataPathKind.Operation && path1.LastSegment.Identifier.TrimEnd(parenthesis).EndsWith(countIdentifier, StringComparison.OrdinalIgnoreCase)))
```

and declare a static readonly char[] property that contains ( and )

Reply:
Following the comment above, does this still stand?

Reply:
no if we assume the above is resolved with no changes, this one can be resolved too.